test/utils_test.cc

```diff
@@ -119,3 +119,9 @@ TEST(UtilsTest, ExpandPath_ShouldReturnReplacedPathIfStartWithTilde) {
   auto replaced = utils::ExpandPath("~/a.txt");
   EXPECT_TRUE(replaced.find("~") == std::string::npos);
 }
+
+TEST(UtilsTest, SpiltWithSymbol_ShouldReturnInputLineListIfHavePipeSymbol) {
+  std::vector<std::string> expected = {"ls ", " > a.txt"};
+  auto inout_line = utils::SpiltWithSymbol("ls | > a.txt", "|");
```

Comment:
This case is not in our scope. Maybe you can try `ls -l | grep 'hello'` which is in our AC.